infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/StreamsSnapshotWriter.java

```diff
@@ -352,6 +320,7 @@ private void applyShadowStream(UUID streamId, long snapshot) {
         if (streamId.equals(REGISTRY_TABLE_ID)) {
             // Only keep entries that currently not exist in registry table
             smrEntries = fetchNewEntries(new ArrayList<>(smrEntries));
```

Comment:
In fetchNewEntries(), we seem to be ignoring the entry if the key is already present. 
Then, if a table previously had` is_federated = false`, and then after an upgrade has` is_federated = true`, the incoming update for the table would be dropped?

Or did I understand it wrong?

Reply:
I didn't follow ... do you refer to the case where a table at Source has `is_federated = true` while at Sink it has `is_federated = false`? In this case, yes, its entries will be dropped